pkg/cmd/experimental/syncgroups/grouppruner.go

```diff
@@ -0,0 +1,84 @@
+package syncgroups
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+)
+
+// GroupPruner runs a prune job on Groups
+type GroupPruner interface {
+	Prune() (errors []error)
+}
+
+// LDAPGroupPruner prunes Groups referencing records on an external LDAP server
+type LDAPGroupPruner struct {
+	// Lists all groups to be synced
+	GroupLister interfaces.LDAPGroupLister
+	// Fetches a group and extracts object metainformation and membership list from a group
+	GroupLocator interfaces.LDAPGroupLocator
+	// Maps an LDAP group enrty to an OpenShift Group's Name
+	GroupNameMapper interfaces.LDAPGroupNameMapper
+	// Allows the Syncer to search for OpenShift Groups
+	GroupClient client.GroupInterface
+	// Host stores the address:port of the LDAP server
+	Host string
+	// DryRun indicates that no changes should be made.
+	DryRun bool
+
+	// Out is used to provide output while the sync job is happening
+	Out io.Writer
+	Err io.Writer
+}
+
+// Prune allows the LDAPGroupSyncer to be a GroupSyncer
+func (s *LDAPGroupPruner) Prune() ([]string, []error) {
+	var groupsToPrune []string
+	var errors []error
+
+	// determine what to sync
+	glog.V(1).Infof("LDAPGroupPruner listing groups to prune with %v", s.GroupLister)
+	ldapGroupUIDs, err := s.GroupLister.ListGroups()
+	if err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+	glog.V(1).Infof("LDAPGroupPruner will attempt to prune ldapGroupUIDs %v", ldapGroupUIDs)
+
+	for _, ldapGroupUID := range ldapGroupUIDs {
+		glog.V(1).Infof("Checking LDAP group %v", ldapGroupUID)
+
+		// get membership data
+		exists, err := s.GroupLocator.Exists(ldapGroupUID)
+		if err != nil {
+			fmt.Fprintf(s.Err, "Error determining LDAP group membership for %q: %v.\n", ldapGroupUID, err)
+			errors = append(errors, err)
+			continue
+		}
+		if !exists {
+			// if the LDAP entry that was previously used to create the group doesn't exist, prune it
+			groupName, err := s.GroupNameMapper.GroupNameFor(ldapGroupUID)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+
+			if !s.DryRun {
+				fmt.Fprintf(s.Out, "group/%s\n", groupName)
```

Comment:
It's conditional on `sync-groups`... we have the opposite of this inside of the `Run()` method for the `Options` structs to print when there is a dry run.